localstack-core/localstack/services/lambda_/invocation/lambda_models.py

```diff
@@ -168,15 +169,17 @@ def _download_archive_to_file(self, target_file: IO) -> None:
         )
         target_file.flush()
 
-    def generate_presigned_url(self, endpoint_url: str | None = None) -> str:
+    def generate_presigned_url(self, endpoint_url: str) -> str:
         """
         Generates a presigned url pointing to the code archive
         """
-        s3_client = connect_to(
+        s3_client = boto3.client(
+            "s3",
             region_name=AWS_REGION_US_EAST_1,
             aws_access_key_id=config.INTERNAL_RESOURCE_ACCOUNT,
+            aws_secret_access_key=INTERNAL_AWS_SECRET_ACCESS_KEY,
             endpoint_url=endpoint_url,
-        ).s3
+        )
```

Comment:
Good thinking adding the `INTERNAL_AWS_SECRET_ACCESS_KEY` 🚀 Pre-signed URLs, due to their nature, are pretty tricky and they will only pass if the secret access key is `test` or the credentials used are "existing" in LocalStack and the secret key id can be retrieved from the Access Key 👍 the fallback to using the default secret key for any access key id was added in #10265